charts/agent-control/charts/agent-control-deployment/templates/uninstall-job.yaml

```diff
@@ -1,29 +1,30 @@
 {{- if .Values.cleanupManagedResources -}}
-{{- $uninstallJobName := include "newrelic.common.naming.truncateToDNSWithSuffix" ( dict "name" (include "newrelic.common.naming.fullname" .) "suffix" "deployment-uninstall" ) -}}
+{{- $uninstallName := include "newrelic.common.naming.truncateToDNSWithSuffix" ( dict "name" (include "newrelic.common.naming.fullname" .) "suffix" "deployment-uninstall" ) -}}
+{{- $uninstallServiceAccountName := include "newrelic.common.naming.truncateToDNSWithSuffix" (dict "name" (include "newrelic.common.serviceAccount.name" .) "suffix" "deployment-uninstall" ) -}}
+
 {{- /*
 The resources managed by the agent-control and the label selector are hardcoded on the agent-control.
+Notice that instrumentations needs to go first since it depends on a chart installed through helmreleases
 */ -}}
-{{- $saCRList := (list "helmreleases.helm.toolkit.fluxcd.io" "helmrepositories.source.toolkit.fluxcd.io") -}}
-{{- $saResourcesLabelSelector := "app.kubernetes.io/managed-by=newrelic-agent-control" -}}
-{{- /*
-To understand why this job installs manifests instead of using Helm, read the comment at `job-manifests.yaml`.
-*/ -}}
+{{- $acCRList := (list "instrumentations.newrelic.com" "helmreleases.helm.toolkit.fluxcd.io" "helmrepositories.source.toolkit.fluxcd.io") -}}
+{{- $acResourcesLabelSelector := "app.kubernetes.io/managed-by=newrelic-agent-control" -}}
+
 apiVersion: batch/v1
 kind: Job
 metadata:
   annotations:
     helm.sh/hook: pre-delete
     helm.sh/hook-delete-policy: before-hook-creation,hook-succeeded
     helm.sh/hook-weight: "10010"
-  name: {{ $uninstallJobName }}
+  name: {{ $uninstallName }}
   namespace: {{ .Release.Namespace }}
 spec:
   ttlSecondsAfterFinished: 120
   backoffLimit: 3
   template:
     spec:
       restartPolicy: Never
-      serviceAccountName: {{ include "newrelic.common.serviceAccount.name" . }}
+      serviceAccountName: {{ $uninstallServiceAccountName }}
```

Comment:
for a moment i thought this was going to cause issues in the future when we add new resources but we still need to manually add the removal of those in this script until we apply the ownership mechanism. So is good to have separate rbac 👍